test/unit/gateways/blue_snap_test.rb

```diff
@@ -297,6 +299,24 @@ def test_does_not_send_level_3_when_empty
     assert_success response
   end
 
+  def test_fraud_response_handling
+    @gateway.expects(:raw_ssl_request).returns(fraudulent_purchase_response)
+
+    response = @gateway.purchase(@fraudulent_amount, @credit_card, @options)
+    assert_failure response
+    assert_match(/fraud-reference-id/, response.message)
+    assert_match(/fraud-event/, response.message)
+  end
+
+  def test_fraud_response_handling_multiple_triggers
+    @gateway.expects(:raw_ssl_request).returns(fraudulent_purchase_response_multiple_triggers)
+
+    response = @gateway.purchase(@fraudulent_amount, @fraudulent_card, @options)
+    assert_failure response
+    assert_match(/orderTotalDecline/, response.message)
+    assert_match(/blacklistPaymentCountryDecline/, response.message)
+  end
+
```

Comment:
Beyond the scope of this task, and maybe beyond the scope of any task, but for some reason, the way the responses are stubbed below causes `response.message` to contain empty elements that it doesn't contain in the remote test: 

```
[303, 312] in /Users/ruthanfreese/dev/active_merchant/test/unit/gateways/blue_snap_test.rb
   303:     @gateway.expects(:raw_ssl_request).returns(fraudulent_purchase_response)
   304: 
   305:     response = @gateway.purchase(@fraudulent_amount, @credit_card, @options)
   306:     assert_failure response
   307:     require 'pry-byebug'; byebug
=> 308:   end
   309: 
   310:   def test_fraud_response_handling_multiple_triggers
   311:     @gateway.expects(:raw_ssl_request).returns(fraudulent_purchase_response_multiple_triggers)
   312: 
(byebug) response.message
"{\"text\":\"\\n      \",\"fraud-reference-id\":\"6270209\",\"fraud-event\":[{\"fraud-event-code\":\"orderTotalDecline\",\"fraud-event-decision\":\"D\",\"fraud-event-expression\":\"3729 > 3728\"}]}"
```

```
[176, 185] in /Users/ruthanfreese/dev/active_merchant/test/remote/gateways/remote_blue_snap_test.rb
   176:   def test_fraudulent_purchase
   177:     # Reflects specific settings on Bluesnap sandbox account.
   178:     response = @gateway.purchase(@fraudulent_amount, @fraudulent_card, @options)
   179:     assert_failure response
   180:     require 'pry-byebug'; byebug
=> 181:     assert_match(/fraud-reference-id/, response.message)
   182:     assert_match(/fraud-event/, response.message)
   183:   end
   184: 
   185:   def test_failed_purchase
(byebug) response.message
"{\"fraud-reference-id\":\"6270313\",\"fraud-event\":[{\"fraud-event-code\":\"blacklistPaymentCountryDecline\",\"fraud-event-decision\":\"D\",\"fraud-event-expression\":\"BR is in list: [BR]\"},{\"fraud-event-code\":\"orderTotalDecline\",\"fraud-event-decision\":\"D\",\"fraud-event-expression\":\"3729 > 3728\"}]}"
```

Reply:
Yep. It's just a weird thing. 🤷‍♀